test/recipes/70-test_tls13messages.t

```diff
@@ -296,4 +301,61 @@ checkhandshake($proxy, checkhandshake::HRR_RESUME_HANDSHAKE,
                | checkhandshake::PSK_CLI_EXTENSION
                | checkhandshake::PSK_SRV_EXTENSION,
                "Resumption handshake with HRR test");
+
+#Test 16: signature_algorithms with 1.3-only ClientHello
+my $testtype = PURE_SIGALGS;
+my $dsa_status = 1;
+$proxy->clear();
+$proxy->clientflags("-tls1_3");
+$proxy->filter(\&modify_sigalgs_filter);
+$proxy->start();
+ok($dsa_status, "DSA/SHA2 sigalg sent for 1.3-only ClientHello");
+
+#Test 17: signature_algorithms with backwards compatible ClientHello
+$testtype = COMPAT_SIGALGS;
+$dsa_status = 1;
+$proxy->clear();
+$proxy->filter(\&modify_sigalgs_filter);
+$proxy->start();
+ok($dsa_status, "DSA sigalg not sent for compat ClientHello");
 unlink $session;
+
+sub modify_sigalgs_filter
+{
+    my $proxy = shift;
+
+    # We're only interested in the initial ClientHello
+    return if ($proxy->flight != 0);
+
+    foreach my $message (@{$proxy->message_list}) {
+        my $ext;
+        my @algs;
+
+        if ($message->mt == TLSProxy::Message::MT_CLIENT_HELLO) {
+            if ($testtype == PURE_SIGALGS) {
+                $ext = $message->extension_data->{TLSProxy::Message::EXT_SIG_ALGS};
+                @algs = unpack('S>*', $ext);
+                # unpack will unpack the length as well
+                shift @algs;
+                foreach (@algs) {
+                    if ($_ == 0x0402 || $_ == 0x0502 || $_ == 0x0602) {
```

Comment:
I guess they could.  This line would probably have to be split in two as the namespacing would expand the lengths, but that's not the end of the world.